### PR TITLE
[ENH] 753 sparkles create a role password validator

### PR DESCRIPTION
**Pull request by CChemin**
Signed-off-by: Sommer Le <pierrick.le-sommer@insee.fr>

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -112,6 +112,7 @@ Sugoi-api implements spring security with full customization allowed
 | fr.insee.sugoi.api.regexp.role.admin                       |                                                                                                                        |               |         |
 | fr.insee.sugoi.ldap.default.group_manager_source_pattern                | Default pattern to use when searching manager group for application. Application name should be passed via {appliname} |               |         |
 | fr.insee.sugoi.api.regexp.role.password.manager            |                                                                                                                        |               |         |
+| fr.insee.sugoi.api.regexp.role.password.validator          |                                                                                                                        |               |         |
```
Comment (CChemin) [inline docs/configuration.md:115]: Je propose : | fr.insee.sugoi.api.regexp.role.password.validator    | Default pattern to use when searching password manager in a realm or userstorage. Userstorage should be passed as\$(realm) and userstorage as\$(userstorage) | no default   |   PASSWORD\_\VALIDATOR\_\$(realm),PASSWORD\_VALIDATOR\_\$(realm)\_$(userstorage)  |

File: sugoi-api-rest-services/src/main/java/fr/insee/sugoi/services/decider/AuthorizeMethodDecider.java [hunk 2]
```diff
@@ -89,6 +89,25 @@ public boolean isPasswordManager(String realm, String userStorage) {
     return true;
   }
 
+  public boolean isPasswordValidator(String realm, String userStorage) {
+    if (enable) {
+      logger.info(
+          "Check if user is at least reader on realm {} and userStorage {}", realm, userStorage);
```
Comment (CChemin) [inline sugoi-api-rest-services/src/main/java/fr/insee/sugoi/services/decider/AuthorizeMethodDecider.java:95]: typo : password validator et pas reader

File: sugoi-api-core/src/test/java/fr/insee/sugoi/core/service/PermissionServiceTests.java [hunk 3]
```diff
@@ -66,6 +66,21 @@ public void testReaderRegexp() {
...
+    SugoiUser sugoiUser =
+        new SugoiUser("password_validator_realm1", List.of("role_PASSWORD_VALIDATOR_realm1_us1"));
+
+    assertThat(
+        "User can be password validator on realm1",
+        permissions.isPasswordValidator(sugoiUser, "realm1", "us1"),
+        is(true));
+    assertThat(
+        "User cannot be password validator on realm2",
+        permissions.isReader(sugoiUser, "realm2", "us1"),
```
Comment (CChemin) [inline sugoi-api-core/src/test/java/fr/insee/sugoi/core/service/PermissionServiceTests.java:80]: permissions.isPasswordValidator plutôt non ?

**Review by Liamtoha — APPROVED** [review]
> 753-sparkles-create-a-role-password-validator

**Review by Liamtoha — DISMISSED** [review]

**Review by clement-dufaure — APPROVED** [review]

**Pull request merged**